### PR TITLE
spatial: use generic lexical sorting routine

### PR DESCRIPTION
**Pull request by jbduncan**
This PR introduces a modernization change that is not necessarily related to #617 but which was spun out from my attempts at generifying the leaves of `graph`. Benchmarks to follow.

Blocked by https://github.com/gonum/gonum/pull/1940 and https://github.com/gonum/gonum/pull/1953.

### CODE REVIEW

**Comment by jbduncan** [comment]
When checking out this PR and using IntelliJ's "Call Hierarchy" on `BySliceValues`, I see that only tests use it directly or indirectly. Furthermore, the [only additional commit](https://github.com/gonum/gonum/pull/1955/commits/9e82cf86492a8212c09d9e266d88d0536a7b204d) introduced in this PR over https://github.com/gonum/gonum/pull/1940 and https://github.com/gonum/gonum/pull/1953 affects just `TestDo` in `spatial/vptree/vptree_test.go`; #1940 only has GitHub Actions changes; and [I already checked](https://github.com/gonum/gonum/pull/1953#issuecomment-1986234415) that #1953 has no benchmarks to run. Therefore, this PR has no benchmarks to run.

**Comment by jbduncan** [comment]
As discussed in https://github.com/gonum/gonum/pull/1956#issuecomment-1988326786, I'm more than happy to double check if new benchmarks are needed first.

**Comment by kortschak** [comment]
This needs to pick up the changes in master.

**Comment by jbduncan** [comment]
> This needs to pick up the changes in master.

Done. I will pick up the latest changes again if https://github.com/gonum/gonum/pull/1953 is merged in.

File: internal/sorted/doc.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package sorted provides common sorting functions.
+package sorted // import "gonum.org/v1/gonum/internal/ordered"
```
Comment (kortschak) [inline internal/sorted/doc.go:6]: Why is the package name changed?

**Comment by jbduncan** [comment]
As per https://github.com/gonum/gonum/pull/1953#issuecomment-2016622004, I'm closing this PR in favour of putting it in #1953.

File: internal/sorted/doc.go (hunk 1 above)
Comment (jbduncan) [reply internal/sorted/doc.go:6]: See https://github.com/gonum/gonum/pull/1953#discussion_r1536703590

**Pull request closed without merging**